pythonz/downloader.py

```diff
@@ -79,3 +81,13 @@ def fetch(cls, url, filename):
         except IOError:
             sys.stdout.write('\n')
             raise DownloadError('Failed to fetch %s from %s' % (filename, url))
+
+
+def validate_sha256(filename, sha256sum):
+    if sha256sum is not None:
+        with open(filename, 'rb') as f, mmap(f.fileno(), 0, access=ACCESS_READ) as m:
```

Comment:
I get an expection here:

~~~~
Traceback (most recent call last):
  File "pythonz/commands/install.py", line 97, in run_command
    p.install()
  File "pythonz/installer/pythoninstaller.py", line 198, in install
    self.download_and_extract()
  File "pythonz/installer/pythoninstaller.py", line 217, in download_and_extract
    self.download()
  File "pythonz/installer/pythoninstaller.py", line 107, in download
    if not validate_sha256(self.download_file, self.expected_sha256):
  File "pythonz/downloader.py", line 88, in validate_sha256
    with open(filename, 'rb') as f, mmap(f.fileno(), 0, access=ACCESS_READ) as m:
AttributeError: __exit__
~~~~